specs/psl-sql-conversion.md

```diff
@@ -0,0 +1,459 @@
+  "authorId" text NOT NULL  ,
+  PRIMARY KEY ("id", "authorId")
+);
+```
+
+### Mapping a model with singular scalar fields
+
+Prisma Schema Example:
+```groovy
+model Blog {
+    id        String @id @default(cuid())
+    title     String
+    viewCount Int
+    isPremium Boolean
+}
+```
+
+A model in the Prisma schema always maps to a table in the underlying SQL database. Each scalar field is mapped to a column by mapping the field type to a type in the SQL database. [TODO: Insert Reference for the Type Mapping Reference here]
+
+Resulting SQL Example (Postgres):
+```sql
+CREATE TABLE "test"."Blog" (
+  "id" text NOT NULL  ,
+  "title" text NOT NULL DEFAULT '' ,
+  "viewCount" integer NOT NULL DEFAULT 0 ,
+  PRIMARY KEY ("id")
+);
+```
+
+### Mapping a model with scalar list fields
+
+Prisma Schema Example:
+```groovy
+model User {
+    id Int @id
+    ints Int[]
+}
+```
+
+A scalar list field in PSL is mapped to a separate table for the values. The name follows the `Model_fieldname` convention. The important columns contained are nodeId for the specific node a value is associated with and value for the actual value. Position is generated and should allow fancy operations like ordered inserts, but is not used. 
+
+Resulting SQL Example (Postgres):
+```sql
+CREATE TABLE "test"."User" (
+  "id" SERIAL,
+  PRIMARY KEY ("id")
+);
+
+CREATE TABLE "test"."User_ints" (
+  "nodeId" integer   REFERENCES "test"."User"("id") ON DELETE CASCADE,
+  "position" integer NOT NULL  ,
+  "value" integer NOT NULL  ,
+  PRIMARY KEY ("nodeId","position")
+);
+```
+
+### Mapping a model with a unique field
+
+```groovy
+model Blog {
+    id        String @id @default(cuid())
+    title     String
+    slug      String @unique
+}
+```
+
+For a field that is annotated with `@unique` a unique index is created in the underlying database.
+
+Resulting SQL Example (Postgres):
+```sql
+CREATE TABLE "test"."Blog" (
+  "id" text NOT NULL  ,
+  "slug" text NOT NULL DEFAULT '' ,
+  "title" text NOT NULL DEFAULT '' ,
+  PRIMARY KEY ("id")
+);
+
+CREATE UNIQUE INDEX "Blog.slug" ON "test"."Blog"("slug")
+```
+
+### Mapping a model with a compound unique field
+
+```groovy
+model User {
+    id Int @id
+    firstname String
+    lastname String
+    @@unique([firstname, lastname], name: "test")
+}
+```
+
+For the block attribute `@@unique` a unique index is created in the underlying database.
+
+Resulting SQL Example (Postgres):
+```sql
+CREATE TABLE "test"."User" (
+  "firstname" text NOT NULL DEFAULT '' ,
+  "id" SERIAL,
+  "lastname" text NOT NULL DEFAULT '' ,
+  PRIMARY KEY ("id")
+);
+
+CREATE UNIQUE INDEX "test" ON "test"."User"("firstname","lastname")
+```
+
+
+## Mapping Enums
+
+```groovy
+model User {
+    id Int @id
+    enum TestEnum
+}
+
+enum TestEnum{
+  A
+  B
+}
+```
+
+We handle the validation of Enum values in the core, and place no constraints on the db column.  
+
+```sql
+CREATE TABLE "test"."User" (
+  "enum" text NOT NULL DEFAULT 'A' ,
```

Comment:
The `DEFAULT 'A'` doesn't have a direct counterpart in the schema. If this is the behaviour we want, I think we should explain why.

Reply:
I think this is temporary behavior of the migration engine. At the moment we always set a default value for required columns. @mavilein can probably explain what to do about this. But I agree, this is confusing since we have an explicit @default value annotation that is not used here.